code-collab-app/src/editorpage.js

```diff
@@ -32,70 +30,68 @@ class Editorpage extends React.Component {
     if (!cellContents) {
       cellContents = '';
     }
-
-    this.setState({
-      cellText: this.state.cellText.concat(cellContents),
-      cellLocked: this.state.cellLocked.concat(false),
-    });
-
-    this.setState({
-      cellCount: this.state.cellCount + 1,
-    });
+    this.props.peer_data.add_new_cell(cellContents);
+    // TODO - maybe call peer_data method to update other peers
   }
 
   // Locks a cell so the user cannot edit it
   // key: the index of the cell to lock
   lockEditorCell = (key) => {
-    let temp = this.state.cellLocked;
-    temp[key] = true;
-    this.setState({ cellLocked:temp });
+    // Lock cell so user may not edit
+    this.props.peer_data.set_cell_locked(key, true);
   }
 
-  // Unlocks a cell so the user may edit it
+  // Unlock a cell so the user may edit it
   // key: the index of the cell to unlock
   unlockEditorCell = (key) => {
-    let temp = this.state.cellLocked;
-    temp[key] = false;
-    this.setState({ cellLocked:temp });
+    this.props.peer_data.set_cell_locked(key, false);
```

Comment:
Typo? This is how it is spelled in peer_data.js

Also we need to update the cell contents before we release the lock, but based on your response it sounds like we will end up doing this on the peer_data side.

```suggestion
    this.props.peer_data.set_cell_lock(key, false);
```

Reply:
Yup, this will be corrected in the next commit as well. Thank you!